crates/ark/src/data_explorer/histogram.rs

```diff
@@ -0,0 +1,579 @@
+        .try_into()?;
+
+    let values = unwrap!(results.get("values"), None => {
+        return Err(anyhow!("Something went wrong when computing `values`"));
+    });
+    let values_formatted = format_string(values.sexp, format_options);
+
+    let counts: Vec<i32> = unwrap!(results.get("counts"), None => {
+        return Err(anyhow!("Something went wrong when computing `counts`"));
+    })
+    .clone()
+    .try_into()?;
+
+    let other_count = if counts.len() == params.limit as usize {
+        let val: i32 = unwrap!(results.get("other_count"), None => {
+            return Err(anyhow!("Something went wrong when computing `others_count`"))
+        })
+        .clone()
+        .try_into()?;
+        Some(val as i64)
+    } else {
+        None
+    };
+
+    Ok(ColumnFrequencyTable {
+        values: values_formatted,
+        counts: counts.into_iter().map(|v| v as i64).collect(),
+        other_count,
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use harp::assert_match;
+    use harp::environment::R_ENVS;
+    use harp::eval::r_parse_eval0;
+    use harp::object::RObject;
+
+    use super::*;
+    use crate::test::r_test;
+
+    fn default_options() -> FormatOptions {
+        FormatOptions {
+            large_num_digits: 2,
+            small_num_digits: 4,
+            max_integral_digits: 7,
+            thousands_sep: Some(",".to_string()),
+            max_value_length: 100,
+        }
+    }
+
+    fn test_histogram(code: &str, num_bins: i64, bin_edges: Vec<&str>, bin_counts: Vec<i64>) {
+        let column = r_parse_eval0(code, R_ENVS.global).unwrap();
+
+        let hist = profile_histogram(
+            column.sexp,
+            &ColumnHistogramParams {
+                method: ColumnHistogramParamsMethod::Fixed,
+                num_bins: Some(num_bins),
+                quantiles: None,
+            },
+            &default_options(),
+        )
+        .unwrap();
+
+        assert_eq!(hist, ColumnHistogram {
+            bin_edges: bin_edges.into_iter().map(|v| v.to_string()).collect(),
+            bin_counts,
+            quantiles: vec![]
+        })
+    }
+
+    fn test_histogram_sturges(code: &str, bin_edges: Vec<&str>, bin_counts: Vec<i64>) {
+        let column = r_parse_eval0(code, R_ENVS.global).unwrap();
+
+        let hist = profile_histogram(
+            column.sexp,
+            &ColumnHistogramParams {
+                method: ColumnHistogramParamsMethod::Sturges,
+                num_bins: None,
+                quantiles: None,
+            },
+            &default_options(),
+        )
+        .unwrap();
+
+        assert_eq!(hist, ColumnHistogram {
+            bin_edges: bin_edges.into_iter().map(|v| v.to_string()).collect(),
+            bin_counts,
+            quantiles: vec![]
+        })
+    }
+
+    fn test_quantiles<T>(code: &str, quantiles: Vec<f64>, expected: T)
+    where
+        RObject: From<T>,
+    {
+        let column = r_parse_eval0(code, R_ENVS.global).unwrap();
```

Comment:
Might be worth adding `harp::parse_eval_global()`

Reply:
Done in https://github.com/posit-dev/ark/pull/480

By the way, if you don't need the global environments (to assign into or do search path lookups), better evaluate in the base env. This way readers don't have to wonder what the global env is used for and this eliminates a range of possible side effects that could contaminate other tests. I've also added `harp::parse_eval_base()` for this.